components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java

```diff
@@ -0,0 +1,640 @@
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_ADD_REQUEST_INVALID;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_ALREADY_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_DELETE_REQUEST_REQUIRED;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_GET_REQUEST_INVALID;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_REPLACE_REQUEST_INVALID;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_ALREADY_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_NAME_REQUIRED;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_SEARCH_REQUEST_INVALID;
+import static org.wso2.carbon.identity.configuration.mgt.core.util.ConfigurationUtils.generateUniqueID;
+import static org.wso2.carbon.identity.configuration.mgt.core.util.ConfigurationUtils.handleClientException;
+import static org.wso2.carbon.identity.configuration.mgt.core.util.ConfigurationUtils.handleServerException;
+
+/**
+ * Resource Manager service implementation.
+ */
+public class ConfigurationManagerImpl implements ConfigurationManager {
+
+    private static final Log log = LogFactory.getLog(ConfigurationManagerImpl.class);
+    private static final String CONFIGURATION_DAO = "configurationDAOs";
+    private List<ConfigurationDAO> configurationDAOS;
+
+    public ConfigurationManagerImpl(ConfigurationManagerConfigurationHolder configurationManagerConfigurationHolder) {
+
+        this.configurationDAOS = configurationManagerConfigurationHolder.getConfigurationDAOS();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Resources getTenantResources(Condition searchCondition) throws ConfigurationManagementException {
+
+        validateSearchRequest(searchCondition);
+        Resources resources = getConfigurationDAO().getTenantResources(searchCondition);
+        if (resources == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No resources found for the search.");
+            }
+            throw ConfigurationUtils.handleClientException(ErrorMessages.ERROR_CODE_RESOURCES_DOES_NOT_EXISTS, null);
```

Comment:
Since the second parameter should be a String and the method does an empty String validation we can also use StringUtils.EMPTY